hybridse/src/udf/dynamic_lib_manager.cc

```diff
@@ -45,14 +45,25 @@ base::Status DynamicLibManager::ExtractFunction(const std::string& name, bool is
         }
     }
     if (!so_handle) {
-        // use abs path to avoid dlopen search failed in yarn mode
-        char abs_path_buff[PATH_MAX];
-        if (realpath(file.c_str(), abs_path_buff) == NULL) {
-            return {common::kExternalUDFError, "can not get real path " + file};
+        void* handle = dlopen(file.c_str(), RTLD_LAZY);
+        if (handle == nullptr) {
+            std::string err_msg;
+            err_msg = "can not open the dynamic library: " + file + ", error: " + dlerror() + ", try to use abs path";
```

Comment:
dlerror may return NULL, simple wrap it !

Reply:
https://linux.die.net/man/3/dlopen dlsym may be null, so it needs to call dlerror before, but dlopen is ok, see the example